generators/model/templates/customModel.xml

```diff
@@ -0,0 +1,187 @@
+     which is defined further on in the document -->
+<model name="<%-namespacePrefix%>:<%-modelName%>" xmlns="http://www.alfresco.org/model/dictionary/1.0">
+
+  <!-- Optional meta-data about the model -->
+  <description><%-modelDescription%></description>
+  <author><%-modelAuthor%></author>
+  <version><%-modelVersion%></version>
+
+  <!-- Imports are required to allow references to definitions in other models -->
+  <imports>
+    <!-- Import Alfresco Dictionary Definitions -->
+    <import uri="http://www.alfresco.org/model/dictionary/1.0" prefix="d"/>
+    <!-- Import Alfresco Content Domain Model Definitions -->
+    <import uri="http://www.alfresco.org/model/content/1.0" prefix="cm"/>
+  </imports>
+
+  <!-- Introduction of new namespaces defined by this model -->
+  <namespaces>
+    <namespace uri="<%-namespaceUri%>" prefix="<%-namespacePrefix%>"/>
+  </namespaces>
+
+  <!--
+    More details on Constraints can be found at the below links
+    https://wiki.alfresco.com/wiki/Content_Model_Constraints
+    https://docs.alfresco.com/5.1/concepts/metadata-model-contraints.html
+  -->
+  <constraints>
+    <!-- Example LIST Constraint -->
+    <!--
+    <constraint name="<%-namespacePrefix%>:productListConstraint" type="LIST">
+      <parameter name="allowedValues">
+        <list>
+          <value>Foobar Widget</value>
+          <value>Flux Capacitor</value>
+          <value>Big Bold App</value>
+        </list>
+      </parameter>
+    </constraint>
+    -->
+
+    <!-- 
+      Example REGEX Constraint that matches 2 capital letters.
+      The default value for 'requiresMatch' is true. Hence, it need not
+      be mentioned in the below example
+    -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:twoCapitalLetterConstraint" type="REGEX">
+        <parameter name="expression">
+          <value>^[A-Z]{2}$</value>
+        </parameter>
+        <parameter name="requiresMatch">
+          <value>true</value>
+        </parameter>
+      </constraint>
+    -->
+
+    <!-- Example MINMAX constraint -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:monthConstraint" type="MINMAX">
+        <parameter name="minValue">
+          <value>1</value>
+        </parameter>
+        <parameter name="maxValue">
+          <value>12</value>
+        </parameter>
+     </constraint>
+    -->
+
+    <!-- Example LENGTH constraint -->
+    <!--
+      <constraint name="<%-namespacePrefix%>:descriptionConstraint" type="LENGTH">
+        <parameter name="minLength">
+          <value>16</value>
+        </parameter>
+        <parameter name="maxLength">
+          <value>256</value>
+        </parameter>
+     </constraint>
+    -->
+  </constraints>
+
+  <!--
+    Dont forget to add the custom types in Share configuration to expose these types in Share.
+    https://docs.alfresco.com/5.0/tasks/dev-extensions-content-models-tutorials-share-config.html
+  -->
+  <types>
```

Comment:
There are some elements for types and aspects you don't show (the importance is to include them because order matters, don't need to write comments about them per se.) Things like required/enforced, mandatory, facet, multiple and index.